### ISSUE TITLE
dotnet core cannot load FIPS built shared library in linux

### ISSUE THREAD

**Issue by winnieryl** [dotnet/runtime#94598]
### Description

I have a fips built shared lib(.so)in linux and wants to load it in dotnet using DllImport, then I run the program, but it fails:
root@ubuntu2204:~#  dotnet test.dll
799305f152dc771a4474edaa481efcbe8bd5f29f
root@ubuntu2204:~# echo $?
0

### Reproduction Steps

ENV: 
ubuntu 2204, 
dotnet version:  6.0.24

build a shared library using fipsld from openssl, import it in dotnet.

### Expected behavior

The lib can be loaded successfully

### Actual behavior

dotnet exits with random chars output

**Comment by Jozkee** [comment]
Can you share instructions in how to build the library with fipsld?

**Comment by winnieryl** [comment]
@Jozkee  thanks for the reply, here is the link on building using fipsld: 
https://wiki.openssl.org/index.php/Fipsld_and_C%2B%2B
https://stackoverflow.com/questions/58930081/compile-application-with-fips-capable-openssl

**Comment by winnieryl** [comment]
UPDATE:
I just found out that this is related to fips signgature, when I specify FIPS_SIG variable and point to incore, the built library will be signed by incore(invoked by fipsld), dotnet will failed to load this kind of library, when the library is not signed, dotnet is able to load the library. I guess dotnet is unable to parse the .so file when there is fips signgature in it?

**Comment by winnieryl** [comment]
Hi @Jozkee, do you have any new info to share?

**Comment by Jozkee** [comment]
No, sorry I haven't looked into this issue yet. cc @vcsjones in case he can look into it first.

**Comment by vcsjones** [comment]
Though this touches on OpenSSL I don't know if this is a an actual issue for the security folks, as it seems to be more related to loading a shared object with `DllImport` that has a FIPS signature.

**Comment by winnieryl** [comment]
@vcsjones thanks for the comment, not sure if you can find the right folk for this issue ?
